packages/db/convex/discord_node.ts

```diff
@@ -137,8 +140,36 @@ You can search for answers using [search.convex.dev](https://search.convex.dev),
 Please note that community support is available here, and avoid tagging staff unless specifically instructed. Thank you!`,
     );
 
+    const actionRow = new ActionRowBuilder<ButtonBuilder>().addComponents(
+      new ButtonBuilder()
+        .setCustomId("resolveThread")
+        .setLabel("Mark as resolved")
+        .setStyle(ButtonStyle.Success),
+    );
+
     await thread.send({
       embeds: [embed],
+      components: [actionRow],
+    });
+  },
+});
+
+export const resolveThread = action({
+  args: {
+    discordThreadId: v.string(),
+  },
+  handler: async (ctx, { discordThreadId }) => {
+    const thread = await ctx.runQuery(
+      internal.discord.getThreadByDiscordThreadId,
+      {
+        discordThreadId,
+      },
+    );
+    if (!thread) {
+      throw new Error(`Thread ${discordThreadId} not found in database`);
+    }
+    await ctx.runMutation(internal.discord.resolveThread, {
+      threadId: thread._id,
```

Comment:
move this to an `apiMutation` in `discord.ts` - it doesn't need to do anything action-related, and then you can pass in the API key so it can't be called by anyone